File: app/src/main/java/com/haishinkit/MainActivity.java
package com.haishinkit;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.hardware.Camera;

import com.haishinkit.media.AudioInfo;
import com.haishinkit.rtmp.RTMPConnection;
import com.haishinkit.rtmp.RTMPStream;
import com.haishinkit.view.CameraView;

public class MainActivity extends AppCompatActivity {
    private CameraView cameraView = null;
    private RTMPConnection connection = null;
    private RTMPStream stream = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        connection = new RTMPConnection();
        stream = new RTMPStream(connection);

        cameraView = new CameraView(this);
        stream.attachAudio(new AudioInfo());

        Camera camera = cameraView.getCamera();
        Camera.Parameters parameters = camera.getParameters();
        parameters.setPreviewSize(320, 240);
        parameters.setPreviewFrameRate(30);
        camera.setParameters(parameters);

        //stream.attachCamera(cameraView.getCamera());
        setContentView(cameraView);

        connection.connect("rtmp://192.168.179.3/live");
        stream.publish("live");
    }
}

File: app/src/main/java/com/haishinkit/app/MainActivity.kt
package com.haishinkit.app

import android.os.Build
import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.haishinkit.graphics.PixelTransformFactory
import com.haishinkit.app.R
import com.haishinkit.vulkan.VkPixelTransform

class MainActivity : AppCompatActivity(), BottomNavigationView.OnNavigationItemSelectedListener {
    private var fragment: Fragment? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            PixelTransformFactory.registerPixelTransform(VkPixelTransform::class)
        }

        setContentView(R.layout.activity_main)

        val navigation = findViewById<BottomNavigationView>(R.id.navigation)
        navigation.setOnNavigationItemSelectedListener(this)

        fragment = CameraTabFragment.newInstance()
        val transaction = supportFragmentManager.beginTransaction()
        transaction.replace(R.id.content, fragment as Fragment).commit()
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.navigation_home -> {
                fragment = CameraTabFragment.newInstance()
            }
            R.id.navigation_mediaprojection -> {
                fragment = MediaProjectionTabFragment.newInstance()
            }
            R.id.navigation_playback -> {
                fragment = PlaybackTabFragment.newInstance()
            }
            R.id.navigation_dashboard -> {
                fragment = PreferenceTagFragment.newInstance()
            }
            else -> {
            }
        }
        val transaction = supportFragmentManager.beginTransaction()
        transaction.replace(R.id.content, fragment as Fragment).commit()
        return true
    }
}
